File: musical-instruments-store-master/app/static/js/product.js


function getProductById(id,onSuccess){
    const query = new AV.Query('Product');
    query.include('title');
    query.include('description');
    query.include('detail');
    query.include('price');
    query.get(id).then((product) => {
        onSuccess(product)
    });
}



async function createProduct(detail, title, description, price) {
    if (detail !== '') {
        // document.getElementById('right').innerHTML=detail;

        const Title = AV.Object.extend('Strings');
        const c_title = new Title();
        c_title.set('english', title);
        /*    c_title.save().then((t) => {
          console.log('保存成功c。objectId：'+t.getObjectId());
        }, (error) => {
              console.log("error");
        });*/
        const Price = AV.Object.extend('Price');
        const c_price = new Price();
        c_price.set('dollar', price);
        c_price.set('CNY', price * 6);
        /*    c_price.save().then((p) => {
          console.log('保存成功p。objectId：'+p.getObjectId());
        }, (error) => {
              console.log("error");
        });*/
        const DES = AV.Object.extend('Strings');
        const des = new DES();
        des.set('english', description);
        /*    des.save().then((d) => {
          console.log('保存成功d。objectId：'+d.getObjectId());
        }, (error) => {
              console.log("error");
        });*/
        const HTMLs = AV.Object.extend('HTMLs');
        const html = new HTMLs();
        html.set('englishHTML', detail.toString());
        /*   html.save().then((html) => {
          console.log('保存成功htmll。objectId：'+html.getObjectId());
        }, (error) => {
              console.log("error");
          // 异常处理
        });*/
        const Product = AV.Object.extend('Product');
        const product = new Product();
        product.set('price', c_price);
        product.set('title', c_title);
        product.set('detail', html);
        product.set('description', des);
        return await product.save()
//     product.save().then((pro) => {
//   console.log('保存成功pro。objectId：'+pro.getObjectId());
//   return pro
// }, (error) => {
//       console.log("error");
//   // 异常处理
// });
    } else {
        console.log("enter your page");
    }
}

// createProduct('<p>s</p>','ab','sa',15);





function updateProduct(id,detail,title,description,price){
      //  id = $('#id').val();
    const Product = AV.Object.extend('Product');
    const product = Product.createWithoutData(id);
    product.set('title',title);
    product.set('price',price);
    product.set('detail',detail);
    product.set('description',description);
    product.save();

}

function updateEnglishProduct(product,title,description,detail){
    product.get('title').set('english',title);
    product.get('description').set('english',description);
    product.get('detail').set('englishHTML',detail);
    product.save();
}



function updateChineseProduct(product,title,description,detail){
    product.get('title').set('chinese',title);
    product.get('description').set('chinese',description);
    product.get('detail').set('chineseHTML',detail);
    product.save();
}

function setProductCover(product){
    if (document.getElementById('fileField').files.length !== 0) {
        const file = new AV.File(document.getElementById('fileField').files[0].name, document.getElementById('fileField').files[0]);
        product.set('cover', file)
    }
    // file.save().then((file) => {
    //     // insertImgFn(file.get("url"))
    //     const product=new AV.Object.createWithoutData('Product',product_id)
    //     console.log('emmmm')
    //     product.set('cover',file)
    //     product.save().then(()=>{
    //         console.log('success!!!')
    //     })
    // }, (error) => {
    //     // 保存失败，可能是文件无法被读取，或者上传过程中出现问题
    // });
}

function setProductCategory(product_id,categories){
    getCategoryByProduct(product_id,function (maps) {
        AV.Object.destroyAll(maps).then(
          function (deletedObjects) {
              const l=[]
              categories.forEach(function (c,i,a) {
                    const map=new AV.Object("ProductCategoryMap")
                    const product = AV.Object.createWithoutData('Product', product_id);
                    const category = AV.Object.createWithoutData('ProductCategory', c);
                    map.set("category",category)
                    map.set("product",product)
                  l.push(map)
              })
              AV.Object.saveAll(l).then(
                  function (savedObjects) {
                    // 成功保存所有对象时进入此 resolve 函数，savedObjects 是包含所有 AV.Object 的数组
                      console.log("success!!")
                  },
                  function (error) {
                    // 只要有一个对象保存错误就会进入此 reject 函数
                  }
                );

          },
          function (error) {
            // 只要有一个对象删除错误就会进入此 reject 函数
          }
    );
    })

}


function getCategoryByProduct(product_id,onSuccess){
    const product = AV.Object.createWithoutData('Product', product_id);

    // 构建 StudentCourseMap 的查询
    const query = new AV.Query('ProductCategoryMap');

    // 查询所有选择了线性代数的学生
    query.equalTo('product', product);

    // 执行查询
    query.find().then(function (maps) {
        // studentCourseMaps 是所有 course 等于线性代数的选课对象
        // 然后遍历过程中可以访问每一个选课对象的 student,course,duration,platform 等属性
        // maps.forEach(function (scm, i, a) {
        //     const student = scm.get('student');
        //     const duration = scm.get('duration');
        //     const platform = scm.get('platform');
        // });
        onSuccess(maps)
    });
}


function getProductByVisitCount( onSuccess,skip=0, limit=10) {


    const query = new AV.Query('Product')
    query.descending('visit_count')
    query.include('title')
    query.include('description')
    query.include('price')
    query.limit(limit)
    query.skip(skip)
    query.find().then((products)=>{
        onSuccess(products)
    })
}

File: musical-instruments-store-master/app/utils/product.py
import string
import leancloud
from .. import models
import config

if __name__ == '__main__':
    leancloud.init("pPObpvTV7pQB9poQHO1NJoMP-MdYXbMMI", "pShwYQQ4JVfSStc56MvkHNrr")


def getProductById(product_id: string, record=False):
    """
    use product_id to get product
    return:  a product av object
    example operation: get the product's information
    product.get('title').get('english') 获取产品英文名
    product.get('title').get('chinese') 获取产品中文名
    product.get('description').get('english') 获取产品英文描述
    product.get('description').get('chinese') 获取产品中文描述
    product.get('price').get('dollar') 获取美元价格
    product.get('price').get('CNY') 获取人民币价格
    product.get('cover').url 获取封面图片链接
    """
    leancloud.User.set_current(config.CURRENT_USER)
    Product = models.Product
    query = Product.query
    query.include('title')
    query.include('description')
    query.include('detail')
    query.include('price')
    product = query.get(product_id)
    if record:
        product.set('visit_count', product.get('visit_count') + 1)
        # print(leancloud.User.get_current())
        leancloud.User.set_current(config.CURRENT_USER)
        # if leancloud.User.get_current()==None:
        #     print('emmmmmmmmmmmmmmmmmmmmm')
        #     username = 'webserver2'
        #     password = '12345678'
        #     user = leancloud.User()
        #     user.login(username=username, password=password)
        product.save()
    return product


def getOrderById(order_id: string, record=False):
    """
    use product_id to get product
    return:  a product av object
    example operation: get the product's information
    product.get('title').get('english') 获取产品英文名
    product.get('title').get('chinese') 获取产品中文名
    product.get('description').get('english') 获取产品英文描述
    product.get('description').get('chinese') 获取产品中文描述
    product.get('price').get('dollar') 获取美元价格
    product.get('price').get('CNY') 获取人民币价格
    product.get('cover').url 获取封面图片链接
    """
    leancloud.User.set_current(config.CURRENT_USER)
    Order = models.Order
    query = Order.query
    query.include('address')
    query.include('email')
    query.include('name')
    query.include('price')
    query.include('product')
    query.include('town')
    query.include('product.title')
    query.include('product.user')
    order = query.get(order_id)
    # if record:
    #     product.set('visit_count', product.get('visit_count') + 1)
    #     # print(leancloud.User.get_current())
    #     leancloud.User.set_current(config.CURRENT_USER)
    #     # if leancloud.User.get_current()==None:
    #     #     print('emmmmmmmmmmmmmmmmmmmmm')
    #     #     username = 'webserver2'
    #     #     password = '12345678'
    #     #     user = leancloud.User()
    #     #     user.login(username=username, password=password)
    #     product.save()
    return order




def getProductByCategory(category_id: string, skip=0, limit=10):
    """
    use category_id to get products
    return:  a list of product av object

    example operation: get the first product's information
    result[0].get('title').get('english') 获取产品英文名
    result[0].get('title').get('chinese') 获取产品中文名
    result[0].get('description').get('english') 获取产品英文描述
    result[0].get('description').get('chinese') 获取产品中文描述
    result[0].get('price').get('dollar') 获取美元价格
    result[0].get('price').get('CNY') 获取人民币价格
    result[0].get('cover').url 获取封面图片链接
    """
    leancloud.User.set_current(config.CURRENT_USER)
    Category = leancloud.Object.extend('ProductCategory')
    category = Category.create_without_data(category_id)
    query = leancloud.Query('ProductCategoryMap')
    query.equal_to('category', category)
    query.include('product')
    query.include('product.title')
    query.include('product.description')
    query.include('product.price')
    query.limit(limit)
    query.skip(skip)
    result = query.find()
    lst = []
    for i in result:
        lst += [i.get('product')]
    return lst


def getProductByVisitCount( skip=0, limit=10):
    """
    use category_id to get products
    return:  a list of product av object

    example operation: get the first product's information
    result[0].get('title').get('english') 获取产品英文名
    result[0].get('title').get('chinese') 获取产品中文名
    result[0].get('description').get('english') 获取产品英文描述
    result[0].get('description').get('chinese') 获取产品中文描述
    result[0].get('price').get('dollar') 获取美元价格
    result[0].get('price').get('CNY') 获取人民币价格
    result[0].get('cover').url 获取封面图片链接
    """
    # Category = leancloud.Object.extend('ProductCategory')
    # category = Category.create_without_data(category_id)
    leancloud.User.set_current(config.CURRENT_USER)
    query = leancloud.Query('Product')
    # query.equal_to('category', category)
    query.descending('visit_count')
    query.include('title')
    query.include('description')
    query.include('price')
    query.limit(limit)
    query.skip(skip)
    result = query.find()
    return result


def getAllCategory(skip=0, limit=50):
    """
    get all category
    return: a list of categories
    example operation: get the first category information
    result[0].get('title').get('english') 获取类别英文名
    result[0].get('title').get('chinese') 获取类别中文名
    result[0].get('cover').url 获取封面图片链接
    """
    leancloud.User.set_current(config.CURRENT_USER)
    query = leancloud.Query('ProductCategory')
    query.limit(limit)
    query.skip(skip)
    query.include('title')
    query.include('img')
    result = query.find()
    return result


def getAllProduct(skip, limit):
    """
    get all product
    return: a list of products
    example operation: get the first category information
    result[0].get('title').get('english') 获取类别英文名
    result[0].get('title').get('chinese') 获取类别中文名
    result[0].get('cover').url 获取封面图片链接
    """
    query = leancloud.Query('Product')
    query.limit(limit)
    query.skip(skip)
    query.include('title')
    query.include('price')
    query.include('description')
    result = query.find()
    return result


def getCategoryById(category_id: string):
    """
    use category_id to get product
    return:  a category av object
    example operation: get the first category information
    result[0].get('title').get('english') 获取类别英文名
    result[0].get('title').get('chinese') 获取类别中文名
    result[0].get('cover').url 获取封面图片链接
    """
    Category = leancloud.Object.extend('ProductCategory')
    query = Category.query
    query.include('title')
    query.include('img')
    category = query.get(category_id)
    return category


def getCategoryByProduct(product_id: string):
    Product = leancloud.Object.extend('Product')
    product = Product.create_without_data(product_id)
    query = leancloud.Query('ProductCategoryMap')
    query.equal_to('product', product)
    query.include('category')
    query.include('category.title')
    result = query.first()
    return result
